lib/dom/dom.nit

```diff
@@ -67,3 +67,27 @@ redef class XMLStartTag
 		abort
 	end
 end
+
+redef class XMLAttrTag
+
+	# Attributes as a map (ignoring malformed attributes)
+	#
+	# ~~~
+	# var xml = """
+	# <student first="Snow" last="Man"/>
+	# """.to_xml
+	#
+	# var attributes = xml["student"].first.as(XMLAttrTag).attributes_map
+	# assert attributes.join(", ", ":") == "first:Snow, last:Man"
+	# ~~~
+	fun attributes_map: Map[String, String]
```

Comment:
I would use a name that suggests that the map is a copy and is recreated each time. Maybe `attributes_to_map` as `to` assume there is a copy&conversion whereas nothing (or `as`) assume a view.